avro4s-core/src/main/scala/com/sksamuel/avro4s/AvroInputStream.scala

```diff
@@ -101,6 +101,12 @@ object AvroInputStream {
   def binary[T: SchemaFor : FromRecord](path: String): AvroBinaryInputStream[T] = binary(Paths.get(path))
   def binary[T: SchemaFor : FromRecord](path: Path): AvroBinaryInputStream[T] = binary(path.toFile)
 
+  def binaryEvolve[T: SchemaFor : FromRecord](in: InputStream, writerSchema: org.apache.avro.Schema): AvroBinaryInputStream[T] = new AvroBinaryInputStream[T](in, writerSchema)
+  def binaryEvolve[T: SchemaFor : FromRecord](bytes: Array[Byte], writerSchema: org.apache.avro.Schema): AvroBinaryInputStream[T] = binaryEvolve(new SeekableByteArrayInput(bytes), writerSchema)
+  def binaryEvolve[T: SchemaFor : FromRecord](file: File, writerSchema: org.apache.avro.Schema): AvroBinaryInputStream[T] = binaryEvolve(new SeekableFileInput(file), writerSchema)
+  def binaryEvolve[T: SchemaFor : FromRecord](path: String, writerSchema: org.apache.avro.Schema): AvroBinaryInputStream[T] = binaryEvolve(Paths.get(path), writerSchema)
+  def binaryEvolve[T: SchemaFor : FromRecord](path: Path, writerSchema: org.apache.avro.Schema): AvroBinaryInputStream[T] = binaryEvolve(path.toFile, writerSchema)
```

Comment:
I don't think so because when reading from a file you have the writer schema encodes within the file itself so on the call site the user doesn't need to specify it..